src/pathspec_user_guide.md

```diff
@@ -0,0 +1,444 @@
+---
```

Comment:
General comments for this whole PR:

1. I'd suggest rethinking how you frame what a PathSpec is. You say that a PathSpec is a concept with "two forms". This isn't exactly true. A PathSpec is a specification with an intended meaning, and is language agnostic. There could be PathSpec support in Python, Go, etc. You should frame PathSpecs as just a specification, and it happens to have implemented bindings in Java (not sure about other languages). I'd recommend highlighting the concept and the specification, and treating the Java binding as just an implementation.

2. All your links are full links to the live website. I'd advise against this. You should either be using absolute path links (e.g. `/rest.li/something`) or better yet relative links using the built-in Jekyll `relative_url` plugin (e.g. `{{ 'something' | relative_url }}`. This second approach is best-practice because it takes into account the "base URL" of the website (e.g. `"/"`, `"/rest.li/"`). See [here](https://github.com/linkedin/rest.li/blob/gh-pages-source/src/_includes/head.html#L28) for an example.

3. I see inconsistent formatting of Rest.li's name (e.g. Rest.Li, Restli, RestLi). I'd suggest referencing Rest.li using the standard format: "Rest.li". Can be solved using simple search-and-replace.

Reply:
(1) reworded, thanks!
(2) Changed, thanks!
(3) updated, thanks.